RoboJackets/igvc-software
# Multiclass Semantic Segmentation

# Description

This PR does the following:

- Implements the Multiclass Segmentation model

Fixes #747 

# Testing steps
1. Make sure to have the model downloaded and placed in the `igvc_sandbox/models` folder
    - (You can either [train your own model](https://github.com/RoboJackets/igvc-software/tree/master/igvc_perception/src/multiclass_segmentation) or download the one located in  `Shared Drives > RoboNav > Software > model_training > full_model4_resized > checkpoints > best.pth`)
2. Run `roscore`
3. Run the rosbag file with something like: `rosbag play ~/path/to/file/bag_file.bag -l`
   - Try using bags from the [2021-2022 dataset](https://cloud.robojackets.org/apps/files/?dir=/RoboNav/Software/rosbags/2021-2022) such as the swervi_parking_lot_2022-05-27-19-46-11.bag
5. Run the multiclass with: `roslaunch igvc_perception multiclass_segmentation.launch`
6. Launch RVIZ and add the `/cam/(center/right/left)/segmented/image...` topic
7. Make sure that semantic segmentation output is visualized on the aforementioned topics
    - `/cam/center/segmented/image`, `/cam/right/segmented/image`,  `/cam/left/segmented/image`, etc

# Expected Output (old version on grass course)
- https://www.dropbox.com/s/yuv4jou4ch4uqla/multiclassSegmentation.webm?dl=0

# Self Checklist
